File: samples/javascript/min.js
var selenium = require('selenium-webdriver');

new selenium.Builder()
.usingServer('http://localhost:8080')
.forBrowser("firefox") // TODO: The bindings require the browserName capability to be valid (but they shouldn't and it will be ignored by the remote end anyway)
.withCapabilities({"app" : "%WINDIR%\\notepad.exe"})
.build()
  .then(driver => {
    return driver.findElement(selenium.By.xpath('//TextField')).sendKeys("Hello from JavaScript!")
      .then(_ => driver.findElement(selenium.By.xpath("//Menu[@caption='File']")).click())
      .then(_ => driver.findElement(selenium.By.xpath("//MenuItem[@caption='Exit']")).click())
      .then(_ => driver.findElement(selenium.By.xpath("//PushButton[@caption=\"Don't Save\"]")).click())
      .then(_ => driver.quit());
  });

File: samples/pythontests/min.py
from selenium import webdriver

driver = webdriver.Remote(
  command_executor='http://127.0.0.1:8080',
  desired_capabilities={'app': '%WINDIR%\\notepad.exe'})

textfield = driver.find_element_by_xpath("//TextField")
textfield.clear()
textfield.send_keys("hello from python!")

driver.find_element_by_xpath("//Menu[@caption='File']").click()
driver.find_element_by_xpath("//MenuItem[@caption='Exit']").click()
driver.find_element_by_xpath("//PushButton[@caption=\"Don't Save\"]").click()

driver.quit()
